pydantic/deprecated/json.py

```diff
@@ -70,11 +73,13 @@ def decimal_encoder(dec_value: Decimal) -> Union[int, float]:
 }
 
 
+@deprecated('pydantic_encoder is deprecated, use BaseModel.model_dump instead.')
 def pydantic_encoder(obj: Any) -> Any:
```

Comment:
Are these really replacements? Wouldn't `AnalyzedType(obj).dump_json()` be a better replacement?